ford/utils.py

```diff
@@ -395,7 +395,10 @@ def obj2dict(intObj):
         if hasattr(intObj, "proctype"):
             extDict["proctype"] = intObj.proctype
         if hasattr(intObj, "extends"):
-            extDict["extends"] = intObj.extends
+            if type(intObj.extends) == ford.sourceform.FortranType:
```

Comment:
One little bit of polish:

```suggestion
            if isinstance(intObj.extends, ford.sourceform.FortranType):
```

I'm very slowly trying to switch to using `isinstance` rather than `type() == ` :)

Reply:
Thanks, applied that modification, should I change it for the remainder of the routine aswell, for consistency? Might obscure the intention of the commit somewhat.

Reply:
Thanks! That's alright, I might try a larger cleanup of these routines when I find time